### PR TITLE
snptest: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package snptest. Only available as precompiled binaries

### CODE REVIEW

File: var/spack/repos/builtin/packages/snptest/package.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class Snptest(Package):
+    """SNPTEST is a program for the analysis of single SNP association in
+       genome-wide studies."""
+
+    homepage = "https://mathgen.stats.ox.ac.uk/genetics_software/snptest/snptest.html"
+
+    version('2.5.2', '', url='http://www.well.ox.ac.uk/~gav/resources/snptest_v2.5.2_linux_x86_64_dynamic.tgz', 
+            when='platform=linux', preferred=True)
+    version('2.5.2', '', url='http://www.well.ox.ac.uk/~gav/resources/snptest_v2.5.2_MacOSX_x86_64.tgz', when='platform=darwin')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snptest/package.py]: Version directives don't yet support a `when` option. For now, you'll either have to add the linux version or the macOS version, but you can't add both. Unless you call the version number `2.5.2-linux`, which is a bit of a hack. You also need a checksum.

Comment (akthoma) [inline var/spack/repos/builtin/packages/snptest/package.py]: Whoops on the checksum

File: var/spack/repos/builtin/packages/snptest/package.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+##############################################################################
+from spack import *
+
+
+class Snptest(Package):
+    """SNPTEST is a program for the analysis of single SNP association in
+       genome-wide studies."""
+
+    homepage = "https://mathgen.stats.ox.ac.uk/genetics_software/snptest/snptest.html"
+    url = "http://www.well.ox.ac.uk/~gav/resources/snptest_v2.5.2_linux_x86_64_dynamic.tgz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snptest/package.py]: I just ran a quick test on this and it doesn't seem to substitute the version properly. Let me see if I can fix that.

File: var/spack/repos/builtin/packages/snptest/package.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    version('2.5.2', 'e3f2cc0351f260cf29369dc4f79a660a')
+
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
+        install('snptest_v2.5.2', prefix.bin)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snptest/package.py]: The name of this executable will be version-dependent. Use `'snptest_v{0}'.format(self.version)`.

File: var/spack/repos/builtin/packages/snptest/package.py (hunk 2 above)
Comment (akthoma) [inline var/spack/repos/builtin/packages/snptest/package.py]: I can also add a `url_for_version`

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snptest/package.py]: Fixed by #4902.

**Pull request merged**